### PR TITLE
LAPS

### PR DESCRIPTION
**Pull request by marcom4rtinez**
add laps to main

### CODE REVIEW

**Comment by marcom4rtinez** [comment]
- [ ]  Add Introduction

File: best-practices/content/40_LAPS.tex [hunk 1]
```diff
@@ -0,0 +1,158 @@
+\chapter{Local Administrator Password Solution}
+\section{Einleitung}
+Local Administrator Password Solution (LAPS)
+%What is it
+%Difference to normal (One Admin Password on all, Annoyingly saving password for all, have domain users with local admin permission)
+
```
Comment (marcom4rtinez) [inline best-practices/content/40_LAPS.tex:6]:
```suggestion
Microsofts Local Administrator Password Solution (LAPS) ist ein Passwort-Manager, der die Active Directory verwendet, um Passwörter für lokale Administratorkonten auf allen Windows-Endgeräten zu verwalten und zu rotieren.
```

File: best-practices/content/40_LAPS.tex [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+\subsection{Anforderungen}
+Um LAPS einsetzten zu können, wir ein Active Directory benötigt.
```
Comment (marcom4rtinez) [inline best-practices/content/40_LAPS.tex:8]:
```suggestion
Um LAPS einsetzen zu können, wird eine Active Directory benötigt.
```

File: best-practices/content/40_LAPS.tex [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+
+
+\section{Installation}
+Die Installation ist in drei Schritte unterteilt.
+Als erstes wird LAPS per Group Policy auf allen Windows Geräten installiert.
```
Comment (marcom4rtinez) [inline best-practices/content/40_LAPS.tex:14]: Use enumerate here

File: best-practices/content/40_LAPS.tex [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+Danach wird das Active Directory für LAPS vorbereitet.
+Das Active Directory braucht noch zwei zusätzliche Attribute auf den Computer Objekten.
```
Comment (marcom4rtinez) [inline best-practices/content/40_LAPS.tex:16]:
```suggestion
Das Active Directory braucht zwei zusätzliche Attribute auf den Computer Objekten, für LAPS.
```

File: best-practices/content/40_LAPS.tex [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+\begin{itemize}
+    \item \textbf{ms-Mcs-AdmPwd}: Speichert das Administrator Passwort in Plain-Text.
```
Comment (marcom4rtinez) [inline best-practices/content/40_LAPS.tex:18]:
```suggestion
    \item \textbf{ms-Mcs-AdmPwd}: Speichert das Administrator Passwort in Klartext.
```

File: best-practices/content/40_LAPS.tex [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+\end{itemize}
+
+Zum Schluss wird LAPS per Group Policy aktiviert.
+
+
+\subsection{Softwareverteilung}
+LAPS kann auf der \href{https://www.microsoft.com/download/details.aspx?id=46899}{Webseite von Microsoft}\footnote{Link: https://www.microsoft.com/download/details.aspx?id=46899} heruntergeladen werden.
+Die .msi Datei muss in einem freigegebenen Netzlaufwerk auf dem Domain Controller platziert werden, auf welches alle Windows Geräte Zugriff haben.
+Zum Beispiel: \textbf{C:\textbackslash Windows\textbackslash SYSVOL\textbackslash sysvol\textbackslash <domain>\textbackslash}.
+Dieses Verzeichniss ist Standardmässig auf Domain Controllern freigegeben.\\
```
Comment (marcom4rtinez) [inline best-practices/content/40_LAPS.tex:29]:
```suggestion
Dieses Verzeichnis ist Standardmässig auf Domain Controllern freigegeben.\\
```

File: best-practices/content/40_LAPS.tex [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+
+Mit \textbf{Rechtsklick $\rightarrow$ Edit} kann die neue Group Policy bearbeitet werden.
+Unter \textbf{Computer Configuration $\rightarrow$ Policies $\rightarrow$ Software Settings $\rightarrow$ Software installation} kann mit \textbf{Rechtsklick $\rightarrow$ New $\rightarrow$ Package\dots} eine Datei ausgewählt werden, welche installiert werden soll.
+Hier muss man die zuvor im freigegebenen Netzlaufwerk abgelegte Installationsdatei auswählen.
+\begin{figure}[H]
+    \centering
+    \includegraphics[width=0.7\linewidth]{../img/LAPS/GPO-Edit-Deployment.png}
+    \caption{GPO für LAPS Deployment bearbeiten}
+\end{figure}
+
```
Comment (marcom4rtinez) [inline best-practices/content/40_LAPS.tex:46]: use minipage?

File: best-practices/content/40_LAPS.tex [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+    \centering
+    \includegraphics[width=0.7\linewidth]{../img/LAPS/GPO-Edit-Deployment-2.png}
+    \caption{LAPS Installationsdatei auswählen}
+\end{figure}
+Die Group Policy kann nun geschlossen werden.
+LAPS wird beim Einloggen auf den jeweiligen Geräten installiert.
+
+\subsubsection{Domain Controller}
+Auf dem DC sollte man LAPS von Hand installieren.
+Bei der manuellen Installation müssen die zusätzliche Features installiert werden.
```
Comment (marcom4rtinez) [inline best-practices/content/40_LAPS.tex:57]:
```suggestion
Bei der manuellen Installation müssen die zusätzlichen Features installiert werden.
```

File: best-practices/content/40_LAPS.tex [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+\begin{figure}[H]
+    \centering
+    \includegraphics[width=0.7\linewidth]{../img/LAPS/laps-ui-install-2.png}
+    \caption{LAPS manuelle Installation}
+\end{figure}
+
+
+\subsubsection{LAPS GUI für Admins}
+Über die GPO wird auf den Computern LAPS ohne GUI installiert.
+Mit dem GUI kann man das Passwort von beliebigen Windows Geräten abfragen.\\
```
Comment (marcom4rtinez) [inline best-practices/content/40_LAPS.tex:70]:
```suggestion
Mit dem GUI kann das Passwort von beliebigen Windows Geräten in der Domäne abgefragt werden.\\
```

File: best-practices/content/40_LAPS.tex [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+
+In der Programmliste in den Systemeinstellungen auf LAPS klicken und im Balken auf \textbf{Change} klicken.
```
Comment (marcom4rtinez) [inline best-practices/content/40_LAPS.tex:72]: Kontext, was machen wir jetzt

File: best-practices/content/40_LAPS.tex [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+\begin{figure}[H]
+    \centering
+    \includegraphics[width=0.7\linewidth]{../img/LAPS/laps-ui-install-2.png}
+    \caption{LAPS GUI Installieren 2}
+\end{figure}
+
+
+\subsection{Active Directory vorbereiten}
+Die zusätzlichen Extension Attributes können per Powershell hinzugefügt werden.
+Der Auszuführende Benutzer muss dafür ``Schema Admin'' sein.\\
```
Comment (marcom4rtinez) [inline best-practices/content/40_LAPS.tex:89]:
```suggestion
Der ausführende Domänenbenutzer muss dafür ein ``Schema Admin'' sein.\\
```

File: best-practices/content/40_LAPS.tex [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+
+Set-AdmPwdComputerSelfPermission -OrgUnit "<Name der OU>"
+#Resultat:
+Name                 DistinguishedName                                              Status
+----                 -----------------                                              ------
+Clients              OU=Clients,OU=Computers,OU=Lab,DC=ba,DC=lab                    Delegated
+\end{lstlisting}
+
+Nun muss noch eine Active Directory Gruppe erstellt werden.
+Dieser kann man alle Benutzer hinzufügen, welche Zugriff auf die Passwörter bekommen sollen.
```
Comment (marcom4rtinez) [inline best-practices/content/40_LAPS.tex:111]:
```suggestion
Dieser Gruppe kann man alle Benutzer hinzufügen, welche Zugriff auf die Passwörter bekommen sollen.
```

File: best-practices/content/40_LAPS.tex [hunk 13]
```diff
@@ -0,0 +1,158 @@
...
+Unter ``Advanced'' muss die neue Gruppe hinzugefügt werden mit der Berechtigung ``All Extended Rights''.
+Nun muss für LAPS noch die Verknüpfung zwischen dieser Gruppe und der OU erstellt werden.
+Dies kann mit Powershell erledigt werden:
+\begin{lstlisting}
+Set-AdmPwdReadPasswordPermission -OrgUnit "Computers" -AllowedPrincipals Global_LAPS-Admins
+Set-AdmPwdResetPasswordPermission -OrgUnit "Computers" -AllowedPrincipals Global_LAPS-Admins
+\end{lstlisting}
+Die Read Berechtigung erlaubt das Lesen der Passwörter.
+Die Reset Berechtigung erlaubt das setzten eines Ablaufdatums eines Passwortes.
+Die Berechtigung kann nun noch mti Powershell überprüft werden:
```
Comment (marcom4rtinez) [inline best-practices/content/40_LAPS.tex:138]:
```suggestion
Die Berechtigung kann nun noch mit Powershell überprüft werden:
```

**Comment by marcom4rtinez** [comment]
@sevigrimm here some inputs from me

File: best-practices/content/40_LAPS.tex (hunk 7 above)
Comment (marcom4rtinez) [reply best-practices/content/40_LAPS.tex:59]: @sevigrimm

File: best-practices/content/40_LAPS.tex (hunk 10 above)
Comment (marcom4rtinez) [reply best-practices/content/40_LAPS.tex:80]: @sevigrimm

File: best-practices/content/40_LAPS.tex [hunk 14]
```diff
@@ -0,0 +1,258 @@
...
+\end{figure}
+
+In der Einstellung ``Password Settings'' kann man die Passwort Länge und den Zyklus setzten, wie oft das Passwort gewechselt werden soll.\\
+
+Die Group Policy kann nun geschlossen werden.
+
+
+\section{Verwendung}
+\subsection{Mit dem GUI}\label{subsec:laps-gui-usage}
+Das auslesen über das GUI geht nur, wenn das GUI auch installiert wurde.
```
Comment (marcom4rtinez) [inline best-practices/content/40_LAPS.tex:182]:
```suggestion
Das auslesen eines Passwortes über das GUI geht nur, wenn das GUI auch installiert wurde.
```

File: best-practices/content/40_LAPS.tex [hunk 15]
```diff
@@ -0,0 +1,258 @@
...
+Das Passwort wird dann im Feld ``Password'' angezeigt.
+\begin{figure}[H]
+    \centering
+    \includegraphics[width=0.7\linewidth]{../img/LAPS/Laps-ui.png}
+    \caption{LAPS Passwort auslesen}
+\end{figure}
+Im Feld ``New expiration time'' kann man eine neue Zeit setzten, wann das Passwort gewechselt werden soll.
+
+\subsection{Mit Powershell}
+Powershell muss als Benutzer gestartet werden, welcher Berechtigung hat das Passwort anzusehen oder Zurückzusetzten.
```
Comment (marcom4rtinez) [inline best-practices/content/40_LAPS.tex:215]:
```suggestion
Die Powershell muss als Benutzer gestartet werden, welcher die Berechtigung hat das Passwort auszulesen oder zurückzusetzten.
```

File: best-practices/content/40_LAPS.tex [hunk 16]
```diff
@@ -0,0 +1,258 @@
...
+Powershell als andere Benutzer starten kann man gleich wie LAPS im Kapitel \hyperref[subsec:laps-gui-usage]{``Mit dem GUI''} als anderer Benutzer gestartet wurde.\\
```
Comment (marcom4rtinez) [inline best-practices/content/40_LAPS.tex:216]:
```suggestion
Powershell als ein anderer Benutzer starten kann man gleich wie LAPS im Kapitel \hyperref[subsec:laps-gui-usage]{``Mit dem GUI''} als anderer Benutzer gestartet wurde.\\
```

File: best-practices/content/40_LAPS.tex [hunk 17]
```diff
@@ -0,0 +1,258 @@
...
+
+Für das Auslesen und Zurücksetzten gibt es folgende zwei Befehle:
```
Comment (marcom4rtinez) [inline best-practices/content/40_LAPS.tex:218]:
```suggestion
Für das Auslesen und Zurücksetzen gibt es folgende Befehle:
```

File: best-practices/content/40_LAPS.tex [hunk 18]
```diff
@@ -0,0 +1,258 @@
...
+\begin{lstlisting}
+    Get-AdmPwdPassword -ComputerName <Computer Name>
+    Reset-AdmPwdPassword -ComputerName <Computer Name>
+\end{lstlisting}
+
+
+\section{Best Practices}
+Wenn möglich sollten nur IT Mitarbeitende die Berechtigung erhalten, Passwörter auslesen zu können.
```
Comment (marcom4rtinez) [inline best-practices/content/40_LAPS.tex:226]:
```suggestion
Idealerweise sollten nur ausgewählte IT Mitarbeitende die Berechtigung erhalten, Passwörter auslesen zu können.
```

File: best-practices/content/40_LAPS.tex [hunk 19]
```diff
@@ -0,0 +1,258 @@
...
+Alle Lokale Administratorrechte von AD Benutzern sollten entfernt werden und nur noch LAPS Passwörter verwendet werden.
+Dadurch kann man verhindern, dass es einen Benutzer gibt, welcher auf allen Systemen Berechtigung hat.\\
```
Comment (marcom4rtinez) [inline best-practices/content/40_LAPS.tex:228]:
```suggestion
Dadurch kann man verhindern, dass es einen Benutzer gibt, welcher auf allen Systemen berechtigt ist.\\
```

File: best-practices/content/40_LAPS.tex (hunk 7 above)
Comment (sevigrimm) [reply best-practices/content/40_LAPS.tex:59]: done

File: best-practices/content/40_LAPS.tex [hunk 20]
```diff
@@ -0,0 +1,258 @@
...
+    \includegraphics[width=0.7\linewidth]{../img/LAPS/laps-ui-install-2.png}
+    \caption{LAPS manuelle Installation}
+\end{figure}
+
+
+\subsubsection{LAPS GUI für Admins}\label{subsubsec:Laps-Gui}
+Über die GPO wird auf den Computern LAPS ohne GUI installiert.
+Mit dem GUI kann das Passwort von beliebigen Windows Geräten in der Domäne abgefragt werden.\\
+
+In der Programmliste in den Systemeinstellungen auf LAPS klicken und im Balken auf \textbf{Change} klicken.
```
[the remark this answers is not in the document]
Comment (sevigrimm) [reply best-practices/content/40_LAPS.tex:80]: @marcom4rtinez review

**Review by sevigrimm — APPROVED** [review]

**Pull request merged**